pkg/steps/release/promote.go

```diff
@@ -43,11 +43,17 @@ type promotionStep struct {
 	mirrorFunc     func(source, target string, tag api.ImageStreamTagReference, date string, imageMirror map[string]string)
 }
 
-func targetName(config api.PromotionConfiguration) string {
+func targetName(registry string, config api.PromotionConfiguration) string {
 	if len(config.Name) > 0 {
-		return fmt.Sprintf("%s/%s:${component}", config.Namespace, config.Name)
+		if registry == api.QuayOpenShiftCIRepo {
+			return fmt.Sprintf("%s/%s_%s_${component}", registry, config.Namespace, config.Name)
+		}
+		return fmt.Sprintf("%s/%s/%s:${component}", registry, config.Namespace, config.Name)
+	}
+	if registry == api.QuayOpenShiftCIRepo {
+		return fmt.Sprintf("%s/%s_${component}_%s", registry, config.Namespace, config.Tag)
 	}
```

Comment:
You know what I'm going to say about these `if`s =)  In general, "quay" appearing in this file is a layering violation.

We're also duplicating the logic already present in `pkg/api/promotion.go`.

Reply:
Let me try with another way.